make/ci.mk

```diff
@@ -19,6 +19,10 @@
 ## @category CI
 ci-presubmit: verify-imports verify-errexit verify-boilerplate verify-codegen verify-crds verify-modules
 
+.PHONY: verify-golangci-lint
+verify-golangci-lint: test/integration/versionchecker/testdata/test_manifests.tar | $(NEEDS_GOLANGCI-LINT)
+	find . -name go.mod -execdir $(GOLANGCI-LINT) run --timeout=30m --config=$(CURDIR)/.golangci.ci.yaml \;
```

Comment:
self review: The command we currently use in GHA uses `find -printf` which doesn't work on macOS. So I can't copy+paste and run that command locally - hence why this PR uses a different command

Reply:
Great shout, I didn't test with a vendored go. Have updated the command based on https://github.com/cert-manager/cert-manager/blob/7b36cfa80836e1c336ed68b42249b20dc4378520/make/util.mk#L21